page-objects/ownerInformationPage.ts

```diff
@@ -77,4 +77,8 @@ export class OwnerInformationPage extends HelperBase {
     private returnVisitRowByPetAndDescription(petName: string, description: string) {
         return this.page.locator('app-pet-list', { has: this.page.getByText(petName) }).locator('app-visit-list').locator('tr', { has: this.page.getByText(description) })
     }
+
+    async returnTotalVisit(petName: string) {
+        return this.page.locator('app-pet-list', { has: this.page.getByText(petName) }).getByRole('button', { name: "Edit Visit" })
+    }
```

Comment:
Instead of returning the count of "Edit Visit" buttons using it later for the assertion, it's better to create a method responsible for assertion. For example validateVisitCountByPetName(petName: string, visitCount: number)

This will improve test readability and optimize the code.
Also don't use "Edit Visit" buttons as your counter. While it works, it's technically not correct. You can count the rows of the table visits. Each row represents the visit. 
Imagine developers will make a redesign and change the button name to "Modify Visit". Your test will fail, but not because something is wrong with the number of visits.

Reply:
I can't understand why the next solution doesn't work, cannot find such locators:
```
async validateVisitCountByPetName(petName: string, visitCount: number) {
        await expect(this.page.locator('app-pet-list', { has: this.page.getByText(petName) })
            .locator('app-visit-list tr.ng-star-inserted')).toHaveCount(visitCount)
    }
```
<img width="670" alt="Знімок екрана 2024-08-20 о 22 47 44" src="https://github.com/user-attachments/assets/a426ef6d-6d49-48fb-b486-72b69d2256e6">


Could you please explain why?

Reply:
```
Locator: locator('app-pet-list').filter({ has: getByText('Leonardo') }).locator('app-visit-list tr.ng-star-inserted')
Expected: 10
Received: 0
Call log:
  - expect.toHaveCount with timeout 5000ms
  - waiting for locator('app-pet-list').filter({ has: getByText('Leonardo') }).locator('app-visit-list tr.ng-star-inserted')
  -   locator resolved to 0 elements
  -   unexpected value "0"
  -   locator resolved to 0 elements
  -   unexpected value "0"
  -   locator resolved to 0 elements
  -   unexpected value "0"
  -   locator resolved to 0 elements
  -   unexpected value "0"
  -   locator resolved to 0 elements
  -   unexpected value "0"
  -   locator resolved to 0 elements
  -   unexpected value "0"
  -   locator resolved to 0 elements
  -   unexpected value "0"
  -   locator resolved to 0 elements
  -   unexpected value "0"
  -   locator resolved to 0 elements
  -   unexpected value "0"
```

Reply:
got is, thanks )